cloud/common.go

```diff
@@ -0,0 +1,303 @@
+// nil and error will be returned.
+func (c Client) Request(method, route, body string, headers map[string]string) (*http.Response, error) {
+	req, err := http.NewRequest(method, c.RawURL(route), bytes.NewReader([]byte(body)))
+	if err != nil {
+		return nil, checkSocketError(c.socketPath, err)
+	}
+
+	for h, v := range headers {
+		req.Header.Set(h, v)
+	}
+
+	resp, err := c.socket.Do(req)
+	if err != nil {
+		return nil, checkSocketError(c.socketPath, err)
+	}
+
+	return resp, nil
+}
+
+// RequestRawURL is an alias for Request
+// this is to maintain the same API as weldr.Client
+func (c Client) RequestRawURL(method, route, body string, headers map[string]string) (*http.Response, error) {
+	return c.Request(method, route, body, headers)
+}
+
+// GetJSON sends a GET and sets the Content-Type header set to "application/json"
+// Errors from the API are returned as an error
+func (c Client) GetJSON(path string) ([]byte, error) {
+	headers := map[string]string{"Content-Type": "application/json"}
+	resp, err := c.Request("GET", path, "", headers)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	responseBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	// Pass the body to the callback function
+	c.rawFunc("GET", path, resp.StatusCode, responseBody)
+
+	// Convert the API's JSON error response to an error
+	if resp.StatusCode == 400 || resp.StatusCode == 404 || resp.StatusCode == 500 {
+		return responseBody, fmt.Errorf("GET %s failed with status %d: %s", path, resp.StatusCode, ErrorToString(responseBody))
+	}
+
+	return responseBody, nil
+}
+
+// PostRaw sends a POST with raw data and returns the raw response body
+// Errors from the API are returned as an error
+func (c Client) PostRaw(path, body string, headers map[string]string) ([]byte, error) {
+	resp, err := c.Request("POST", path, body, headers)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	responseBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	// Pass the body to the callback function
+	c.rawFunc("POST", path, resp.StatusCode, responseBody)
+
+	// TODO make sure this covers cloud API errors...
+	// Convert the API's JSON error response to an error
+	if resp.StatusCode == 400 || resp.StatusCode == 404 || resp.StatusCode == 500 {
+		return responseBody, fmt.Errorf("POST %s failed with status %d: %s", path, resp.StatusCode, ErrorToString(responseBody))
+	}
+
+	return responseBody, nil
+}
+
+// PostJSON sends a POST with JSON data and the Content-Type header set to "application/json"
+// Errors from the API are returned as an error
+func (c Client) PostJSON(path, body string) ([]byte, error) {
+	headers := map[string]string{"Content-Type": "application/json"}
+	return c.PostRaw(path, body, headers)
+}
+
+// IsStringInSlice returns true if the string is present, false if not
+// slice must be sorted
+func IsStringInSlice(slice []string, s string) bool {
```

Comment:
Can this be `slices.Contains` instead or was that introduced later than our minimum version?

Reply:
I change that in a followup commit that refactors some common things between cloud and weldr into a common package.